internal/app/machined/pkg/controllers/perf/perf.go

```diff
@@ -0,0 +1,111 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package perf
+
+import (
+	"context"
+	"time"
+
+	"github.com/cosi-project/runtime/pkg/controller"
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/prometheus/procfs"
+	"go.uber.org/zap"
+
+	"github.com/talos-systems/talos/pkg/resources/perf"
+)
+
+// StatsController manages v1alpha1.Stats which is the current snaphot of the machine CPU and Memory consumption.
+type StatsController struct{}
+
+// Name implements controller.StatsController interface.
+func (ctrl *StatsController) Name() string {
+	return "perf.StatsStatsController"
+}
+
+// Inputs implements controller.StatsController interface.
+func (ctrl *StatsController) Inputs() []controller.Input {
+	return nil
+}
+
+// Outputs implements controller.StatsController interface.
+func (ctrl *StatsController) Outputs() []controller.Output {
+	return []controller.Output{
+		{
+			Type: perf.CPUType,
+			Kind: controller.OutputExclusive,
+		},
+		{
+			Type: perf.MemoryType,
+			Kind: controller.OutputExclusive,
+		},
+	}
+}
+
+// Run implements controller.StatsController interface.
+func (ctrl *StatsController) Run(ctx context.Context, r controller.Runtime, logger *zap.Logger) error {
+	ticker := time.NewTicker(time.Second * 5)
+
+	var (
+		fs          procfs.FS
+		procfsReady bool
+		err         error
+	)
+
+	for {
+		select {
+		case <-ctx.Done():
+			ticker.Stop()
+
+			return nil
+		case <-ticker.C:
+			if !procfsReady {
+				fs, err = procfs.NewDefaultFS()
+				if err != nil {
+					logger.Error("failed to init default procfs", zap.Error(err))
```

Comment:
as a side note, you can return `err` here so that controller will be restarted by the runtime. 

this enables exponential backoff, and also we can do some tooling showing failing controllers (vs. failing silently here)

it's more of a idea, as of course the way it is done right now it should work as well as the ticker will force the retry